src/infra/core/src/engine/engine_odf.rs

```diff
@@ -97,26 +129,85 @@ impl ODFEngine {
             }
         })
     }
+
+    pub async fn materialize_response(
+        &self,
+        engine_response: ExecuteQueryResponseSuccess,
+        out_data_path: PathBuf,
+        out_checkpoint_path: PathBuf,
+    ) -> Result<TransformResponse, EngineError> {
+        let out_data = if engine_response.data_interval.is_some() {
+            if !out_data_path.exists() {
+                return Err(EngineError::contract_error(
+                    "Engine did not write a response data file",
+                    Vec::new(),
+                )
+                .into());
+            }
+            if out_data_path.is_symlink() || !out_data_path.is_file() {
```

Comment:
The attack vector I was thinking about is that an engine instead of writing a regular file would write a symlink pointing to something like `~/.config/app/passwords` to attempt to get this file included as data into some public dataset.